spring-xd-dirt/src/test/java/org/springframework/xd/dirt/integration/bus/kafka/KafkaMessageBusTests.java

```diff
@@ -321,6 +322,49 @@ public void testCustomPartitionCountOverridesPartitioningIfLarger() throws Excep
 	}
 
 	@Test
+	public void testNamedChannelPartitionCount() throws Exception {
+
+		byte[] ratherBigPayload = new byte[2048];
+		Arrays.fill(ratherBigPayload, (byte) 65);
+		KafkaTestMessageBus messageBus = (KafkaTestMessageBus) getMessageBus();
+
+		DirectChannel moduleOutputChannel = new DirectChannel();
+		QueueChannel moduleInputChannel = new QueueChannel();
+		Properties producerProperties = new Properties();
+		producerProperties.put(BusProperties.MIN_PARTITION_COUNT, "5");
+		Properties consumerProperties = new Properties();
+		consumerProperties.put(BusProperties.MIN_PARTITION_COUNT, "5");
+		long uniqueBindingId = System.currentTimeMillis();
+		messageBus.bindProducer("queue:foo" + uniqueBindingId + ".0", moduleOutputChannel, producerProperties);
+		messageBus.bindConsumer("queue:foo" + uniqueBindingId + ".0", moduleInputChannel, consumerProperties);
+		producerProperties.put(BusProperties.PARTITION_KEY_EXPRESSION, "payload");
```

Comment:
We may want to remove this particular aspect of the test - and allow in the future using a partition key expression for any Kafka producer - specifically in the light of https://github.com/spring-projects/spring-xd/issues/1852